### PR TITLE
cli: implement list-tracks

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/cli/store.py [hunk 1]
```diff
@@ -728,3 +728,48 @@ def whoami():
...
+    """
+    store_client_cli = StoreClientCLI()
+    snap_channel_map = store_client_cli.get_snap_channel_map(snap_name=snap_name)
+
+    # Iterate over the entries, replace None with - for consistent presentation
+    track_table = (
+        [
+            track.name,
+            track.status,
+            track.creation_date if track.creation_date else "-",
```
Comment (cjp256) [inline snapcraft/cli/store.py:762]: Would the "-" sort to the top of the table?

File: snapcraft/storeapi/v2/channel_map.py [hunk 2]
```diff
@@ -184,6 +184,51 @@ def __init__(
...
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}: {self.name!r}>"
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        status: str,
+        creation_date: str,
```
Comment (cjp256) [inline snapcraft/storeapi/v2/channel_map.py:223]: Should this be `Optional[str]`?

File: snapcraft/cli/store.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/cli/store.py:762]: yes, and to be fair, the only "track" with no creation date is "latest"

File: snapcraft/storeapi/v2/channel_map.py (hunk 2 above)
Comment (sergiusens) [reply snapcraft/storeapi/v2/channel_map.py:223]: Good catch

File: snapcraft/cli/store.py (hunk 1 above)
Comment (cjp256) [reply snapcraft/cli/store.py:762]: sounds good!

**Comment by sergiusens** [comment]
Thanks @cjp256, I switched mostly because it got messy with mypy

**Pull request merged**